clang/test/SemaSYCL/built-in-type-kernel-arg.cpp

```diff
@@ -13,6 +13,8 @@ __attribute__((sycl_kernel)) void kernel(const Func &kernelFunc) {
 struct test_struct {
   int data;
   int *ptr; // Unused pointer in struct
+  int *ptr_array1[2];    // Unused array of pointers in struct
```

Comment:
Do we have a test case when array of pointers struct member is used in kernel?

Reply:
We don't. But the AST would remain the same since the object (test_struct) is captured. If the invalid pointer is used in kernel, FE won't do anything since we don't diagnose pointers in FE.

Reply:
Should I remove the unused comment from these tests since it doesn't mean anything in the FE? I guess it is only really needed for runtime test where we're making sure unused pointers are just ignored (instead of crash/diagnostic)

Reply:
I see. 

> Should I remove the 'unused.....' comment from these tests since it doesn't mean anything in the FE?

I'll leave it is up to you. This comment technically is valid.

That would be great to have a runtime test with *used* array of USM pointers to be paranoid and double check that everything works correctly one day, but it can be done separately.

Reply:
I'm actually not sure if we have any tests (even non-array) for USM pointers inside a struct. We should probably test this functionality. I'll file a new issue to look into this.

Reply:
Filed: https://github.com/intel/llvm/issues/2425

Reply:
> > Should I remove the 'unused.....' comment from these tests since it doesn't mean anything in the FE?
> 
> I'll leave it is up to you. This comment technically is valid.

I removed it since it is confusing.